fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,175 @@
 These are metrics which are not part of `scikit-learn`.
 """
 
+import numpy as np
 import sklearn.metrics as skm
 
 from ._balanced_root_mean_squared_error import _balanced_root_mean_squared_error  # noqa: F401
 from ._mean_predictions import mean_prediction, _mean_overprediction, _mean_underprediction  # noqa: F401,E501
 from ._selection_rate import selection_rate  # noqa: F401,E501
 
+_NEED_TWO_UNIQUE_Y_VALS = "Must have two unique y values"
+_NEED_POS_LABEL_IN_Y_VALS = "Must have pos_label in y values"
 
-def true_positive_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the true positive rate (also called sensitivity, recall, or hit rate)."""
+
+def _get_labels_for_confusion_matrix(y_true, y_pred, pos_label):
+    r"""Figure out the labels argument for skm.confusion_matrix.
+
+    This is an internal method used by the true/false positive/negative
+    rate metrics (and hence are restricted to binary data). We compute
+    these using the confusion matrix.
+    In general, we expect the input data to be :math:`\in {0, 1}`
+    with 1 being the 'positive' label. However, users might want to have
+    a different pair of values, and then specify the positive label
+    for themselves.
+    This method prepares the `labels` argument of
+    :py:func:\sklearn.metrics.confusion_matrix` based on the
+    user's specifications.
+
+    Parameters
+    ----------
+    y_true : array-like
+        The true values for the metric computation
+
+    y_pred : array-like
+        The predicted alues for the metric computation
+
+    pos_label : scalar
+        The value in the true and predicted arrays to treat as positive
+
+    Returns
+    -------
+    list
+        A two element list, consisting of the unique values of the two arrays
+        with the positive label listed last
+    """
+    my_labels = list(np.unique(np.concatenate((y_true, y_pred), axis=None)))
+    if len(my_labels) != 2:
+        raise ValueError(_NEED_TWO_UNIQUE_Y_VALS)
+    if pos_label not in my_labels:
+        raise ValueError(_NEED_POS_LABEL_IN_Y_VALS)
+    if my_labels[1] != pos_label:
+        my_labels = list(reversed(my_labels))
+    return my_labels
+
+
+def true_positive_rate(y_true, y_pred, sample_weight=None, pos_label=1):
```

Comment:
default in all of these should be `pos_label=None`. the meaning of that will be the same as `pos_label=1`, but only if `my_labels` in `_get_labels_for_confusion_matrix` are a subset of `{0, 1}`. otherwise, an exception needs to be raised.

this will prevent a bad corner case, e.g., when the unique labels are `[1, 2]`. in that case defaulting to `pos_label=1` might not be what's intended.

Reply:
Actually, scikit-learn is not consistent:
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html?highlight=precision_score

@adrinjalali do you know why scikit-learn has these different defaults for `pos_label`?

Reply:
My 2c: on a high level, if the labels appearing in data are `[1,2]`, then it seems wrong to default to 1 as a `pos_label`.

Reply:
I don't disagree - however, looking at `recall_score` (which *is* the true positive rate), `pos_label` defaults to 1.

Reply:
Having just experimented with defaulting `pos_label` to `None`, suppose that all the y values are the same....how do we automatically determine the 'positive' value - or do we just raise an error at that point?

Reply:
I'm going with raising an error.

Reply:
I prefer the proposal in https://github.com/fairlearn/fairlearn/pull/467#discussion_r439187689 . I'd say the inconsistencies in sklearn come from the fact that they're not written at the same time and not by the same people, and therefore there are quite a bit of those inconsistencies all around the library. But if I wanna chose one, I'd go or the `None` default.

Reply:
I've reworked the handling, I believe in line with what @MiroDudik  suggested.